arches/app/etl_modules/bulk_data_deletion.py

```diff
@@ -0,0 +1,188 @@
+from datetime import datetime
+import json
+import logging
+import uuid
+from django.contrib.auth.models import User
+from django.db import connection
+from django.http import HttpRequest
+from django.utils.translation import gettext as _
+from arches.app.etl_modules.base_data_editor import BaseBulkEditor
+from arches.app.etl_modules.decorators import load_data_async
+from arches.app.models.resource import Resource
+from arches.app.models.system_settings import settings
+from arches.app.models.tile import Tile
+import arches.app.tasks as tasks
+from arches.app.utils.index_database import index_resources_by_transaction
+
+logger = logging.getLogger(__name__)
+
+
+
+class BulkDataDeletion(BaseBulkEditor):
+    def delete_resources(self, userid, loadid, graphid, resourceids):
+        result = {"success": False}
+        user = User.objects.get(id=userid)
+        try:
+            if resourceids and graphid:
+                resources = Resource.objects.filter(graph_id=graphid).filter(pk__in=resourceids)
+            elif graphid:
+                resources = Resource.objects.filter(graph_id=graphid)
+            elif resourceids:
+                resources = Resource.objects.filter(pk__in=resourceids)
+            for resource in resources.iterator():
+                resource.delete(user=user, index=False, transaction_id=loadid)
+            result["success"] = True
+        except Exception as e:
+            logger.exception(e)
+            result["message"] = _("Unable to delete resources: {}").format(str(e))
+
+        return result
+
+    def delete_tiles(self, userid, loadid, nodegroupid, resourceids):
+        result = {"success": False}
+        user = User.objects.get(id=userid)
+
+        try:
+            if resourceids:
+                tiles = Tile.objects.filter(nodegroup_id=nodegroupid).filter(resourceinstance_id__in=resourceids)
+            else:
+                tiles = Tile.objects.filter(nodegroup_id=nodegroupid)
+            for tile in tiles.iterator():
```

Comment:
I think this would be faster for large operations if we add a chunk size:
```for tile in tiles.iterator(chunk_size=1000):```

Reply:
set the chunk_size = 2000